consensus/src/epoch_manager.rs

```diff
@@ -203,13 +204,15 @@ impl EpochManager {
                     heuristic_config.active_weights,
                     heuristic_config.inactive_weights,
                 ));
-                Box::new(LeaderReputation::new(
+                let pe = Box::new(LeaderReputation::new(
                     epoch_state.epoch,
                     proposers,
                     backend,
                     heuristic,
                     onchain_config.leader_reputation_exclude_round(),
-                ))
+                ));
+                // LeaderReputation is not cheap, so we can cache the amount of rounds round_manager needs.
+                Box::new(CachedProposerElection::new(pe, 3))
```

Comment:
Instead of hardcoding the window size, do we want to make this configurable?

Reply:
value should be based on how things are used. so I can make it constant with comments, but it shouldn't be configurable.

round_manager logic currently access current and previous round. In the stacked diff, when we add failed leaders history, I extend this by the amount of history we access as well.